ADAL/src/urlprotocol/ADURLProtocol.m

```diff
@@ -202,36 +213,38 @@ - (void)connection:(NSURLConnection *)connection
 
 #pragma mark - NSURLConnectionDataDelegate Methods
 
-- (NSURLRequest *)connection:(NSURLConnection *)connection willSendRequest:(NSURLRequest *)request redirectResponse:(NSURLResponse *)response
+- (NSURLRequest *)connection:(NSURLConnection *)connection willSendRequest:(NSURLRequest *)request redirectResponse:(NSURLResponse *)redirectResponse
 {
     (void)connection;
     
-    AD_LOG_VERBOSE_F(@"-[ADURLProtocol connection:willSendRequest:]", _correlationId, @"Redirect response: %@. New request:%@", response.URL, request.URL);
+    AD_LOG_VERBOSE_F(@"-[ADURLProtocol connection:willSendRequest:]", _correlationId, @"Redirect response: %@. New request:%@", redirectResponse.URL, request.URL);
     
     // Disallow HTTP for ADURLProtocol
     if ([request.URL.scheme isEqualToString:@"http"])
     {
-        if (!s_endURL)
+        if ([request.URL.absoluteString.lowercaseString hasPrefix:s_endURL])
```

Comment:
non http/https can't be loaded in Safari anyways, and https URLs don't have the same security implications http ones do.